control/Session.php

```diff
@@ -314,6 +314,23 @@ public static function get($name) {
 	public static function get_all() {
 		return self::current_session()->inst_getAll();
 	}
+	
+	/**
+	 * Get or return variables, depending on method signature:
+	 *
+	 * Session::data()              returns Session::get_all()
+	 * Session::data($name)         returns Session::get($name)
+	 * Session::data($name, $value) returns Session::set($name, $value)
+	 */
+	public static function data() {
+		$args = func_get_args();
+		$num_args = count($args);
```

Comment:
Yeah, thought about that, but since I'd already used `func_get_args`, I figured `count` might make more sense ...